cmd/root/kubernetes/containers.go

```diff
@@ -0,0 +1,206 @@
+//	Copyright 2023 Dremio Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// kubernetes package provides access to log collections on k8s
+package kubernetes
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/dremio/dremio-diagnostic-collector/pkg/simplelog"
+)
+
+// CommandExecutor is an interface that represents the functionality of executing commands.
+type CommandExecutor interface {
+	ExecuteCommand(command string, args ...string) ([]byte, error)
+}
+
+// DefaultCommandExecutor is the default implementation of the CommandExecutor interface.
+type DefaultCommandExecutor struct{}
+
+// ExecuteCommand executes the specified command and returns the output and any error.
+func (e *DefaultCommandExecutor) ExecuteCommand(command string, args ...string) ([]byte, error) {
```

Comment:
on all of this executecommand stuff, we already use the cli.Cli struct to execute everything, I am not sure why we need to reimplement the same functionality here.

Reply:
file now removed completely